packages/auto-label/src/auto-label.ts

```diff
@@ -125,30 +125,6 @@ handler.checkExistingIssueLabels = async function checkExistingIssueLabels(
   }
 };
 
-// maybeAddAPILabel adds the given label if there isn't already an api: label.
-handler.maybeAddAPILabel = async (
-  github: GitHubAPI,
-  owner: string,
-  repo: string,
-  issueNumber: number,
-  labelsOnIssue: Label[] | null,
-  label: string
-): Promise<string | undefined> => {
-  if (labelsOnIssue) {
-    const found = labelsOnIssue.find(element =>
-      element.name.startsWith('api:')
-    );
-    if (found) {
-      console.log(`${owner}/${repo}#${issueNumber} already has an api: label`);
-      return undefined;
-    }
-    await handler.addLabels(github, owner, repo, issueNumber, [label]);
-    return label;
-  }
-  await handler.addLabels(github, owner, repo, issueNumber, [label]);
-  return label;
-};
-
```

Comment:
removed this logic because it essentially replicates what the main logic does: checks to see if the label already exists, and then either does or doesn't add it. I changed the logic below to incorporate the possibility of an auto-detected label. I also added a clean-up function, which will remove any tags that are not either the auto-detected one or assigned by the file.